account-integrations/safe/src/SafeZkEmailRecoveryPlugin.sol

```diff
@@ -94,69 +126,146 @@ contract SafeZkEmailRecoveryPlugin {
-            dkimPublicKeyHash
+            dkimPublicKeyHash,
+            dkimRegistry
         );
-        dkimRegistryOfSafe[safe] = dkimRegistry;
     }
 
     /**
-     * @notice recovers a safe ecdsa plugin using a zk email proof.
-     * @dev rotates the safe ecdsa plugin owner address to a new address. The email address hash is kept hidden until this function is called.
-     * This function is designed so it can be called from any account and account type.
+     * @notice initiates a recovery of a safe ecdsa plugin using a zk email proof.
+     * @dev rotates the safe ecdsa plugin owner address to a new address.
+     * Uses the default delay period if no custom delay has been set.
      * @param safe the safe that manages the safe ecdsa plugin being recovered
-     * @param ecdsaPlugin safe ecsda plugin address that this function will be rotating the owner address for
      * @param newOwner the new owner address of the safe ecdsa plugin
      * @param emailDomain domain name of the sender's email
      * @param a part of the proof
      * @param b part of the proof
      * @param c part of the proof
      */
-    function recoverAccount(
+    function initiateRecovery(
         address safe,
-        address ecdsaPlugin,
         address newOwner,
         string memory emailDomain,
         uint256[2] memory a,
         uint256[2][2] memory b,
         uint256[2] memory c
     ) external {
-        ZkEmailRecoveryStorage memory recoveryStorage = zkEmailRecoveryStorage[
-            safe
-        ];
+        RecoveryRequest memory recoveryRequest = recoveryRequests[safe];
+
+        if (recoveryRequest.executeAfter > 0) {
+            revert RECOVERY_ALREADY_INITIATED();
+        }
 
         if (
             !this.isDKIMPublicKeyHashValid(
                 safe,
                 emailDomain,
-                recoveryStorage.dkimPublicKeyHash
+                recoveryRequest.dkimPublicKeyHash
             )
         ) {
             revert INVALID_DKIM_KEY_HASH(
                 safe,
                 emailDomain,
-                recoveryStorage.dkimPublicKeyHash
+                recoveryRequest.dkimPublicKeyHash
             );
         }
 
         uint256[4] memory publicSignals = [
             uint256(uint160(safe)),
-            uint256(recoveryStorage.recoveryHash),
+            uint256(recoveryRequest.recoveryHash),
             uint256(uint160(newOwner)),
-            uint256(recoveryStorage.dkimPublicKeyHash)
+            uint256(recoveryRequest.dkimPublicKeyHash)
         ];
 
         // verify proof
         bool verified = verifier.verifyProof(a, b, c, publicSignals);
         if (!verified) revert INVALID_PROOF();
 
-        bytes memory data = abi.encodeWithSignature(
-            "enable(bytes)",
-            abi.encodePacked(newOwner)
-        );
+        uint256 delay;
+        if (customDelay[safe] > 0) {
+            delay = customDelay[safe];
+        } else {
+            delay = defaultDelay;
+        }
```

Comment:
Move this into `configureRecovery` and always set the delay for each config to be explicit. You might also want to rename the mapping to something like `recoveryDelay`.

Reply:
makes sense, done